src/main/java/org/betonquest/betonquest/quest/event/spawn/Equipment.java

```diff
@@ -0,0 +1,80 @@
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.item.QuestItem;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Mob;
+import org.bukkit.inventory.EntityEquipment;
+import org.bukkit.persistence.PersistentDataType;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A wrapper for the equipment and drops of an entity.
+ *
+ * @param helmet     the helmet to equip the entity with
+ * @param chestplate the chest plate to equip the entity with
+ * @param leggings   the leggings to equip the entity with
+ * @param boots      the boots to equip the entity with
+ * @param mainHand   the main hand item to equip the entity with
+ * @param offHand    the off-hand item to equip the entity with
+ * @param drops      the items to drop when the entity dies
+ */
+public record Equipment(@Nullable QuestItem helmet, @Nullable QuestItem chestplate,
+                        @Nullable QuestItem leggings, @Nullable QuestItem boots, @Nullable QuestItem mainHand,
+                        @Nullable QuestItem offHand, Item[] drops) {
+
+    /**
+     * Adds the drops to the entity.
+     *
+     * @param entity  the entity to add the drops to
+     * @param profile the profile to get the drop amounts from
+     * @throws QuestRuntimeException if the variable could not be resolved
+     */
+    public void addDrops(final Entity entity, @Nullable final Profile profile) throws QuestRuntimeException {
+        int dropIndex = 0;
+        for (final Item item : drops) {
+            final String value = item.getID().getFullID() + ":" + item.getAmount().getValue(profile).intValue();
+            final NamespacedKey key = new NamespacedKey(BetonQuest.getInstance(), "betonquest-drops-" + dropIndex);
+            entity.getPersistentDataContainer().set(key, PersistentDataType.STRING, value);
+            dropIndex++;
+        }
+    }
+
+    /**
+     * Adds the equipment to the entity.
+     *
+     * @param living the entity to add the equipment to
+     */
+    public void addEquipment(final LivingEntity living) {
+        final EntityEquipment equipment = living.getEquipment();
+        if (equipment != null) {
+            equipment.setHelmet(helmet == null ? null : helmet.generate(1));
+            equipment.setChestplate(chestplate == null ? null : chestplate.generate(1));
+            equipment.setLeggings(leggings == null ? null : leggings.generate(1));
+            equipment.setBoots(boots == null ? null : boots.generate(1));
+            equipment.setItemInMainHand(mainHand == null ? null : mainHand.generate(1));
+            equipment.setItemInOffHand(offHand == null ? null : offHand.generate(1));
+        }
+    }
+
+    /**
+     * Sets the drop chances for the equipment to 0.
+     *
+     * @param mob the mob to remove the equipment drops for
+     */
+    public void removeEquipmentDrops(final Mob mob) {
```

Comment:
Why does `addEquipment` use `LivingEntity` and `removeEquipmentDrops` use `Mob`?

Reply:
> Not a bug introduced by you, but when I spawn an armorstand I get this error (Paper 1.20.4 build 463):
> 
> ```yaml
> events:
>   spawn_clear: 'spawn %location% ARMOR_STAND 1 marked:tex'
> ```
> 
> Before (BQ 2.1.3):
> 
> ```
> Caused by: java.lang.IllegalArgumentException: Cannot set drop chance for non-Mob entity
> ```
Here is a previous present error where you can only set the drop chance of Mobs, not LivingEntittiy.
For the sake of reasonability a comment in the methods doc would be useful.